### DISCUSSION TITLE
Sudden crash when loading beatmap

### DISCUSSION THREAD

**Discussion by NotGumballer91** [ppy/osu#15876] [Q&A] [answered]
This has never happened to me before until 21.11.27. It happens with any kind beatmap.
Unfortunately, I couldn't find a consistent way of reproducing it. Maybe it has something to do with my installed ruleset.

I couldn't catch what the error message was but it was something along lines of "0x000005c" and "Invalid parameters". 
I'll update if happens again.

Here are the logs:
[network.log](https://github.com/ppy/osu/files/7624211/network.log)
[performance.log](https://github.com/ppy/osu/files/7624212/performance.log)
[runtime.log](https://github.com/ppy/osu/files/7624207/runtime.log)
[updater.log](https://github.com/ppy/osu/files/7624209/updater.log)
[database.log](https://github.com/ppy/osu/files/7624210/database.log)
[performance-audio.log](https://github.com/ppy/osu/files/7624203/performance-audio.log)
[performance-draw.log](https://github.com/ppy/osu/files/7624204/performance-draw.log)
[performance-input.log](https://github.com/ppy/osu/files/7624205/performance-input.log)
[performance-update.log](https://github.com/ppy/osu/files/7624206/performance-update.log)

**Comment by smoogipoo** [comment]
Nothing interesting in the logs, unfortunately.

**Comment by NotGumballer91** [comment]
I've found no issues so far. Maybe the custom ruleset was causing it (I've tried it after updating it). So I guess this is no longer an issue.

**Comment by NotGumballer91** [comment]
It crashed after several days of no problem.
I have no idea what causes this but it could be the mod's customization values?
I was trying out different types of mods with some few custom values and it happened.
Video:
https://user-images.githubusercontent.com/64581009/145450099-6d36180f-6d71-48b5-a06b-bc5c98594dbe.mp4
Error message:
![image](https://user-images.githubusercontent.com/64581009/145450564-18dd1f6b-9d63-43ab-867f-3a62c4ac97e3.png)

Logs just in case:
[updater.log](https://github.com/ppy/osu/files/7687397/updater.log)
[database.log](https://github.com/ppy/osu/files/7687399/database.log)
[network.log](https://github.com/ppy/osu/files/7687400/network.log)
[performance.log](https://github.com/ppy/osu/files/7687401/performance.log)
[performance-audio.log](https://github.com/ppy/osu/files/7687402/performance-audio.log)
[runtime.log](https://github.com/ppy/osu/files/7687396/runtime.log)
[performance-input.log](https://github.com/ppy/osu/files/7687394/performance-input.log)
[performance-update.log](https://github.com/ppy/osu/files/7687395/performance-update.log)
[performance-draw.log](https://github.com/ppy/osu/files/7687393/performance-draw.log)

**Reply by smoogipoo** [reply]
Are you somehow running this on Windows 7? Your logs indicate that.

This sort of error usually points to some hardware issue, whether it's bad drivers/bad GPU or bad RAM, or perhaps some broken OS install.

Please provide your [dxdiag output](https://support.microsoft.com/en-us/help/4028644/windows-open-and-run-dxdiagexe).

**Reply by NotGumballer91** [reply]
Strange, I'm not using Windows 7. I was concerned if this was my OS issue.

[DxDiag.txt](https://github.com/ppy/osu/files/7687482/DxDiag.txt)

**Reply by smoogipoo** [reply]
Please make sure you don't have this compatibility mode turned on in the Right click -> Properties menu of the shortcut.

![image](https://user-images.githubusercontent.com/1329837/145452801-434ca7c8-9b54-4f66-9b23-916df4c1c547.png)

**Reply by NotGumballer91** [reply]
Ok, Apparently compatibility mode was set to Windows 7. I've never touched lazer's properties before.

Will this fix the issue?

**Reply by smoogipoo** [reply]
That's up to you to find out, but likely yes. osu!lazer doesn't run on Windows 7 at all (as in, on an actual Windows 7 install), so it's hard to say how compatibility mode may be affecting things.

**Reply by NotGumballer91** [reply]
Alright, I'll update if it doesn't crash or not after few weeks.

**Reply by smoogipoo** [reply]
I'll mark this as answered for now then. :+1:

**Comment by NotGumballer91** [comment]
After 2021.1225.0, It somehow crashed again.
Video:

https://user-images.githubusercontent.com/64581009/147403679-9f94d9b7-15a0-43c6-8de3-d0e3ddbadde1.mp4

As @smoogipoo said, I checked whether compatibility mode was on or not. It wasn't. I'm starting to think it's caused by my custom rulesets. Again, This happens very randomly and I couldn't find a way to replicate it.

Logs: 
[network.log](https://github.com/ppy/osu/files/7776763/network.log)
[performance.log](https://github.com/ppy/osu/files/7776764/performance.log)
[runtime.log](https://github.com/ppy/osu/files/7776765/runtime.log)
[updater.log](https://github.com/ppy/osu/files/7776766/updater.log)
[database.log](https://github.com/ppy/osu/files/7776767/database.log)

**Reply by bdach** [reply]
a native exception like that is more likely to be driver error rather than some custom ruleset. there is no evidence for it being any ruleset's fault in the logs to begin with.

please check windows event viewer and see if there is any more information about this particular crash there.

**Reply by NotGumballer91** [reply]
These what I got:
> Application: osu!.exe
> CoreCLR Version: 5.0.1121.47308
> .NET Version: 5.0.11
> Description: The process was terminated due to an internal error in the .NET Runtime at IP 00007FFA20462BE9 (00007FFA202B0000) with exit code c0000005.

> Description: A .NET application failed.
> Application: osu!.exe
> Path: C:\Users\Gumballer\AppData\Local\osulazer\app-2021.1225.0\osu!.exe
> Message: Cannot use file stream for [C:\Users\Gumballer\AppData\Local\osulazer\app-2021.1225.0\osu!.deps.json]: No such file or directory

Looking further, Apparently I had this problem since June:

> Description: A .NET application failed.
> Application: osu!.exe
> Path: C:\Users\Gumballer\AppData\Local\osulazer\app-2021.619.1\osu!.exe
> Message: Cannot use file stream for [C:\Users\Gumballer\AppData\Local\osulazer\app-2021.619.1\osu!.deps.json]: No such file or directory

Is this something wrong with my lazer?

**Reply by bdach** [reply]
the first one looks most relevant. is there something more there? like a mention of `nvoglv64.dll` or something?

either way driver error, most likely. recommend updating device drivers if you haven't tried yet.

**Reply by NotGumballer91** [reply]
I can't find anything that you've mentioned. There's just nothing there besides that error.
![image](https://user-images.githubusercontent.com/64581009/147453036-2d6849b5-8f54-42a4-9ba3-cbce60c319de.png)

I'll try updating my drivers and see if it happens again or not.

**Comment by NotGumballer91** [comment, marked as answer]
I'm sorry for bumping into this. 
After updating my gpu drivers to 511.23 and 4 months of continuous playing, I've experienced zero crashes.
